client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx

```diff
@@ -0,0 +1,159 @@
+		getProductBySlug( state, productSlug )
+	) as ResponseCartProduct | null;
+
+	const getTimeRemainingForSubscription = ( purchase: Purchase ) => {
+		const purchaseExpiryDate = moment( purchase.expiryDate );
+
+		return moment.duration( purchaseExpiryDate.diff( moment( purchase.mostRecentRenewDate ) ) );
+	};
+
+	const getTimeRemainingTranslatedPeriod = ( purchase: Purchase ) => {
+		const timeRemaining = getTimeRemainingForSubscription( purchase );
+
+		if ( timeRemaining.months() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.months();
+			const periodName = translate( 'month', 'months', { count: timeRemaining.months() } );
+
+			return { timeRemainingNumber, periodName };
+		} else if ( timeRemaining.weeks() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.weeks();
+			const periodName = translate( 'week', 'weeks', { count: timeRemainingNumber } );
+
+			return { timeRemainingNumber, periodName };
+		}
+
+		const timeRemainingNumber = timeRemaining.days();
+		const periodName = translate( 'day', 'days', { count: timeRemainingNumber } );
+
+		return { timeRemainingNumber, periodName };
+	};
+
+	const renderTimeRemainingString = ( product: ResponseCartProduct | null, purchase: Purchase ) => {
+		// returns early if there's no product or accounting for the edge case that the plan expires today (or somehow already expired)
+		// in this case, do not show the time remaining for the plan
+		const timeRemaining = getTimeRemainingForSubscription( purchase );
+		if ( null === product || timeRemaining.days() <= 1 ) {
+			return null;
+		}
+
+		// if this product/ plan is partner managed, it won't really "expire" from the user's perspective
+		if ( isPartnerPurchase( purchase ) || ! purchase.expiryDate ) {
+			return (
+				<React.Fragment>
+					{ translate(
+						'Your {{strong}} %(productName)s {{/strong}} subscription is still active. {{br/}}',
+						{
+							args: { productName: product.product_name },
+							components: {
+								strong: <strong />,
+								br: <br />,
+							},
+						}
+					) }
+				</React.Fragment>
+			);
+		}
+
+		const translatedPeriod = getTimeRemainingTranslatedPeriod( purchase );
+
+		// show how much time is left on the plan
+		return (
+			<React.Fragment>
+				{ translate(
+					'You still have {{strong}}%(timeRemaining)d %(period)s{{/strong}} left on your {{strong}}%(productName)s{{/strong}} subscription. {{br/}}',
+					{
+						args: {
+							timeRemaining: translatedPeriod.timeRemainingNumber,
+							period: translatedPeriod.periodName,
+							productName: product.product_name,
+						},
+						components: {
+							strong: <strong />,
+							br: <br />,
+						},
+						comment:
+							"'period' is either one of 'day', 'week', 'month', or their plural form. 'timeRemaining' is a number representing the time left that will be used with the 'period'.",
+					}
+				) }
+			</React.Fragment>
+		);
+	};
+
+	const getCancelConsequenceByProduct = ( productSlug: string ) => {
```

Comment:
Not a dev task - just a note. Once we have this merged into the test branch, we can ask Rob/ Simon to review these strings. I just made these up 🙂